packages/fxa-settings/src/pages/InlineRecoverySetup/index.tsx

```diff
@@ -0,0 +1,233 @@
+                  </FtlMsg>
+                  <FtlMsg id="inline-recovery-cancel-setup">
+                    <button className="link-blue text-sm">Cancel setup</button>
+                  </FtlMsg>
+                </div>
+              </div>
+            </form>
+          </section>
+        </>
+      ) : (
+        <>
+          <div className="mb-4">
+            {serviceName ? (
+              <FtlMsg id="inline-recovery-setup-header" vars={{ serviceName }}>
+                <h1 className="card-header">
+                  Save backup authentication codes{' '}
+                  <span className="card-subheader">{`to continue to ${serviceName}`}</span>
+                </h1>
+              </FtlMsg>
+            ) : (
+              <FtlMsg id="inline-recovery-setup-header-default">
+                <h1 className="card-header">
+                  Save backup authentication codes{' '}
+                  <span className="card-subheader">
+                    to continue to account settings
+                  </span>
+                </h1>
+              </FtlMsg>
+            )}
+          </div>
+          <section>
+            <FtlMsg id="inline-recovery-setup-message">
+              <p className="text-sm mb-6">
+                Store these one-time use codes in a safe place for when you
+                don’t have your mobile device.
+              </p>
+            </FtlMsg>
+            <div>
+              <div className="flex rounded-xl px-7 font-mono text-center text-sm text-green-900 bg-green-800/10 flex-wrap relative mb-6 py-4">
+                {recoveryCodes.map((code) => {
+                  return <div className="flex-50% py-1">{code}</div>;
+                })}
+              </div>
+              {isIos ? (
+                <div className="flex justify-center w-4/5 max-w-48 mx-auto fill-grey-500">
+                  <button
+                    title={localizedCopyButtonTitle}
+                    className="w-12 h-12 relative inline-block text-grey-500 rounded active:text-blue-600  active:fill-blue-600 focus:outline focus:outline-2 focus:outline-offset-2 focus:outline-blue-500 hover:bg-grey-50"
+                    onClick={copyCode}
+                  >
+                    <CopyIcon
+                      className="mx-auto"
+                      role="img"
+                      aria-label={localizedCopyButtonTitle}
+                    />
+                    {localizedCopyButtonTitle}
+                  </button>
+                </div>
+              ) : (
+                <div className="flex justify-between w-4/5 max-w-48 mx-auto fill-grey-500">
+                  <button
+                    title={localizedDownloadButtonTitle}
+                    className="w-12 h-12 relative inline-block text-grey-500 rounded active:text-blue-600 active:fill-blue-600 focus:outline focus:outline-2 focus:outline-offset-2 focus:outline-blue-500 hover:bg-grey-50 align-center"
```

Comment:
Can we reuse `DataBlock` here? In Storybook the "multiple" state looks a lot like what you've made here:

<img width="606" alt="image" src="https://user-images.githubusercontent.com/13018240/214655655-e0e857fe-f4e6-4f4b-ac6e-dc815184925b.png">

If we can't, since `w-12 h-12 relative inline-block text-grey-500 rounded active:text-blue-600 active:fill-blue-600 focus:outline focus:outline-2 focus:outline-offset-2 focus:outline-blue-500 hover:bg-grey-50` are used commonly across all of these, it seems worth setting these to a `const` in this component and adding the additional classes when needed like ```className={`${recoveryKeyButtonClasses} align-center`}```.

Reply:
Oh mannnnn it's so close!! The only difference is that in our version, if the user is on iOS, they only see the `Copy` action, and not any of the others, whereas `GetDataTrio` is always shown in `DataBlock`.  I would propose adding an optional prop into `DataBlock` to conditionally show only the one `GetData` action?

Reply:
Pushed up a possible solution!! Let me know what you think!